packages/components/src/segmented-control/types.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { MutableRefObject, Ref, ReactNode } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import type { RadioStateReturn } from 'reakit';
+
+/**
+ * Internal dependencies
+ */
+import type { FormElementProps } from '../utils/types';
+
+/**
+ * Option to render within `SegmentedControl`.
+ *
+ * @example
+ * ```jsx
+ * const option = { value: 'elsa', label: 'Elsa' };
+ * ```
+ */
+export type SegmentedControlOption = {
+	value: string | number;
+	label: string;
```

Comment:
Based on the storybook it seems like `label` should be `ReactNode`

Reply:
With `ReactNode ` does it mean that it still could be string or number?

Reply:
Yes! ReactNode includes both string and number 🙂

Reply:
Actually I kept it as a string and change the storybook examples for now. Label is used down the road for `aria-label` which is needed for accessibility. Maybe we could enhance this to accept an Icon in the future? 🤔  -- https://github.com/WordPress/gutenberg/pull/31937/commits/db60768ce147abbf79e5343791526598716ee911

Reply:
Sounds good to me! String labels make more sense anyway. We might also be able to do something with the types to enfore passing `aria-label` if `label` isn't a string like:

```ts
type AriaLabelRequired = {
  label: ReactElement;
  'aria-label': string;
}
type AriaLabelNotRequired = {
  label: ReactText;
}

export type Props = ArialLabelRequired | AriaLabelNotRequired;
```

But just something for the future for us to consider.